Index: OpenRTM-aist/src/lib/rtm/idl/Makefile.am
===================================================================
--- OpenRTM-aist/src/lib/rtm/idl/Makefile.am	(リビジョン 1894)
+++ OpenRTM-aist/src/lib/rtm/idl/Makefile.am	(リビジョン 1895)
@@ -7,7 +7,7 @@
 
 AUTOMAKE_OPTIONS = 1.4
 
-SUBDIRS = device_interfaces
+SUBDIRS = . device_interfaces
 
 IDLC = @IDLC@
 IDLFLAGS = @IDL_FLAGS@
@@ -15,9 +15,6 @@
 AM_CPPFLAGS=-I$(top_srcdir)/src/lib -I$(top_srcdir)/src/lib/rtm
 AM_LDFLAGS=-L$(top_builddir)/src/lib/rtm -L$(top_builddir)/src/lib/rtm/idl
 
-# Suffix definition of sources and headers of skeletons and stubs 
-SUFFIXES = .idl Skel.cpp Stub.cpp Stub.o Skel.o
-
 # RTM idl sources
 IDL_SOURCES =             \
 	SDOPackage.idl        \
@@ -113,6 +110,23 @@
 	$(IDL_SOURCES:.idl=-cpp.h)
 
 
+idl-stamp: $(IDL_SOURCES)
+	@touch idl-stamp
+	@for f in $(IDL_SOURCES) ; do \
+		$(OIDLC) $$f ; \
+		$(IDLC) $(IDLFLAGS) $$f ; \
+		$(top_srcdir)/build/makewrapper.py $$f ; \
+	done ;
+
+$(ALL_SRC): idl-stamp
+	@if test -f $@ ; then \
+		touch $@ ; \
+	else \
+		rm -f idl-stamp ; \
+		$(MAKE) $(AM_MAKEFLAGS) idl-stamp ; \
+	fi ;
+
+
 # Install directory of headers and idl files
 rtmheaderdir = $(prefix)/include/rtm
 idldir =       $(rtmheaderdir)/idl
@@ -122,27 +136,16 @@
 
 nodist_rtmskel_HEADERS = $(ALL_SRC)
 
-
 #------------------------------------------------------------
 # Intermediate library : libRTCSkel.a
 noinst_LTLIBRARIES = libRTCSkel.la libRTCStub.la
 
 nodist_libRTCSkel_la_SOURCES = $(SKEL_CPP)
-
 libRTCSkel_la_LIBADD = $(LIBS)
 
 nodist_libRTCStub_la_SOURCES = $(STUB_CPP)
-
 libRTCStub_la_LIBADD = $(LIBS)
-
-.idlSkel.cpp:
-	$(OIDLC) $<
-	$(IDLC) $(IDLFLAGS) $<
-	$(top_srcdir)/build/makewrapper.py $<
-.idlStub.cpp:
-	$(OIDLC) $<
-	$(IDLC) $(IDLFLAGS) $<
-	$(top_srcdir)/build/makewrapper.py $<
+#
 #------------------------------------------------------------
 
 
@@ -197,6 +200,7 @@
 # Cleanup
 clean-local:
 	rm -f $(ALL_SRC) *~ *.core *.lst Makefile.old
+	rm -f idl-stamp
 
 # Extra sources included distribution
 EXTRA_DIST = \
Index: OpenRTM-aist/examples/SimpleService/Makefile.am
===================================================================
--- OpenRTM-aist/examples/SimpleService/Makefile.am	(リビジョン 1894)
+++ OpenRTM-aist/examples/SimpleService/Makefile.am	(リビジョン 1895)
@@ -81,28 +81,33 @@
 
 dist_MyServiceProvider_la_SOURCES = $(PROVIDER_DIST_SRC) $(PROVIDER_DIST_H) 
 MyServiceProvider_la_LDFLAGS = -module
-MyServiceProvider_la_FLAGS   = -L$(top_builddir)/src/lib/rtm/.libs -L.
-MyServiceProvider_la_LIBADD  = $(top_builddir)/src/lib/rtm/libRTC.la \
-	$(top_builddir)/src/lib/coil/lib/libcoil.la -lMyServiceSkel
+MyServiceProvider_la_DEPENDENCIES = libMyServiceSkel.la
+MyServiceProvider_la_LIBADD  =                \
+	libMyServiceSkel.la                   \
+	$(top_builddir)/src/lib/rtm/libRTC.la \
+	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
 dist_MyServiceConsumer_la_SOURCES = $(CONSUMER_DIST_SRC) $(CONSUMER_DIST_H)
 MyServiceConsumer_la_LDFLAGS = -module
-MyServiceConsumer_la_FLAGS   = -L$(top_builddir)/src/lib/rtm/.libs
-MyServiceConsumer_la_LIBADD  =  -lMyServiceStub \
+MyServiceConsumer_la_DEPENDENCIES = libMyServiceStub.la
+MyServiceConsumer_la_LIBADD  =                \
+	libMyServiceStub.la                   \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
 	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
 # Standalone RTCs
 example_PROGRAMS = MyServiceProviderComp MyServiceConsumerComp
 MyServiceProviderComp_SOURCES = MyServiceProviderComp.cpp
-MyServiceProviderComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
-MyServiceProviderComp_LDADD   = .libs/MyServiceProvider.a \
+MyServiceProviderComp_DEPENDENCIES = MyServiceProvider.la
+MyServiceProviderComp_LDADD   =               \
+	.libs/MyServiceProvider.a             \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
 	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
 MyServiceConsumerComp_SOURCES = MyServiceConsumerComp.cpp
-MyServiceConsumerComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
-MyServiceConsumerComp_LDADD   = .libs/MyServiceConsumer.a \
+MyServiceConsumerComp_DEPENDENCIES = MyServiceConsumer.la
+MyServiceConsumerComp_LDADD   =               \
+	.libs/MyServiceConsumer.a             \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
 	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
Index: OpenRTM-aist/examples/SeqIO/Makefile.am
===================================================================
--- OpenRTM-aist/examples/SeqIO/Makefile.am	(リビジョン 1971)
+++ OpenRTM-aist/examples/SeqIO/Makefile.am	(リビジョン 1972)
@@ -38,12 +38,14 @@
 example_DATA = rtc.conf component.conf
 
 SeqInComp_SOURCES = SeqInComp.cpp 
+SeqInComp_DEPENDENCIES = SeqIn.la
 SeqInComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 SeqInComp_LDADD = SeqIn.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
 	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
 SeqOutComp_SOURCES = SeqOutComp.cpp 
+SeqOutComp_DEPENDENCIES = SeqOut.la
 SeqOutComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 SeqOutComp_LDADD =  SeqOut.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
Index: OpenRTM-aist/examples/SimpleIO/Makefile.am
===================================================================
--- OpenRTM-aist/examples/SimpleIO/Makefile.am	(リビジョン 1971)
+++ OpenRTM-aist/examples/SimpleIO/Makefile.am	(リビジョン 1972)
@@ -39,12 +39,14 @@
 example_DATA = rtc.conf consin.conf consout.conf component.conf rtc.conf.org
 
 ConsoleInComp_SOURCES = ConsoleInComp.cpp 
+ConsoleInComp_DEPENDENCIES = ConsoleIn.la
 ConsoleInComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 ConsoleInComp_LDADD   =  ConsoleIn.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
 	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
 ConsoleOutComp_SOURCES = ConsoleOutComp.cpp 
+ConsoleOutComp_DEPENDENCIES = ConsoleOut.la
 ConsoleOutComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 ConsoleOutComp_LDADD   =  ConsoleOut.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
Index: OpenRTM-aist/examples/Composite/Makefile.am
===================================================================
--- OpenRTM-aist/examples/Composite/Makefile.am	(リビジョン 1971)
+++ OpenRTM-aist/examples/Composite/Makefile.am	(リビジョン 1972)
@@ -44,18 +44,21 @@
 example_DATA = rtc.conf composite.conf
 
 ControllerComp_SOURCES = ControllerComp.cpp 
+ControllerComp_DEPENDENCIES = Controller.la
 ControllerComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 ControllerComp_LDADD   =  Controller.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
 	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
 MotorComp_SOURCES = MotorComp.cpp 
+MotorComp_DEPENDENCIES = Motor.la
 MotorComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 MotorComp_LDADD   = Motor.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
 	$(top_builddir)/src/lib/coil/lib/libcoil.la
 
 SensorComp_SOURCES = SensorComp.cpp 
+SensorComp_DEPENDENCIES = Sensor.la
 SensorComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 SensorComp_LDADD   =  Sensor.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
Index: OpenRTM-aist/examples/ConfigSample/Makefile.am
===================================================================
--- OpenRTM-aist/examples/ConfigSample/Makefile.am	(リビジョン 1971)
+++ OpenRTM-aist/examples/ConfigSample/Makefile.am	(リビジョン 1972)
@@ -33,6 +33,7 @@
 example_DATA     = configsample.conf
 
 ConfigSampleComp_SOURCES = ConfigSampleComp.cpp 
+ConfigSampleComp_DEPENDENCIES = ConfigSample.la
 ConfigSampleComp_LDFLAGS = -L$(top_builddir)/src/lib/rtm/.libs
 ConfigSampleComp_LDADD   = ConfigSample.o \
 	$(top_builddir)/src/lib/rtm/libRTC.la \
Index: OpenRTM-aist/examples/SimpleService/Makefile.am
===================================================================
--- OpenRTM-aist/examples/SimpleService/Makefile.am	(リビジョン 1972)
+++ OpenRTM-aist/examples/SimpleService/Makefile.am	(リビジョン 1973)
@@ -128,7 +128,6 @@
 	rtc.conf
 
 ALL_SRC =                    \
-	$(IDL_SOURCES)       \
         $(PROVIDER_COMP_SRC) \
         $(CONSUMER_COMP_SRC)
 
